### PR TITLE
Null pointers checks

### PR DESCRIPTION
**Pull request by Yugend**
Added some NULL pointer checks.
My static analyzer has indicated these places as dangerous, so I decided to try to fix it. I guess there's not a lot of memory allocated in these places, so these checks will probably never come in handy. But perhaps you will find them not superfluous.

### CODE REVIEW

File: audisp/plugins/zos-remote/zos-remote-queue.c [hunk 1]
```diff
@@ -130,6 +130,11 @@ void increase_queue_depth(unsigned int size)
         void *tmp_q;
 
         tmp_q = realloc(q, size * sizeof(BerElement *));
+        if (tmp_q == NULL) {
+			log_err("Memory allocation error");;
```
[the remark this answers is not in the document]
Comment (Yugend) [reply audisp/plugins/zos-remote/zos-remote-queue.c]: Looks like there is some trouble with my code editor. I just deleted this indentation and added it again, and git showed me, that something has changed :)
Anyway, I have fixed everything you suggested, thanks!

**Comment by stevegrubb** [comment]
Thanks for the patch. I'm not sure I like simple memory error messages. It doesn't give the user anything meaningful to diagnose the issue. In cases like those, it might be better to add 2  %s fields to print __FILE__ and __LINE__. This way we know where it's coming from.

**Comment by Yugend** [comment]
> Thanks for the patch. I'm not sure I like simple memory error messages. It doesn't give the user anything meaningful to diagnose the issue. In cases like those, it might be better to add 2 %s fields to print **FILE** and **LINE**. This way we know where it's coming from.

Thanks for your answer!
I agree, it sounds reasonable. But I was thinking about end users, most likely not all of them may be programmers. Do you think it would be useful for them to get information about which file and which line the error occurred in?.. Perhaps something like "you need more RAM" would be more useful for end users?

For the same reason I replaced 'assert' macros, which only work in debug version of project.

**Comment by stevegrubb** [comment]
I try to keep the memory footprint of auditd low. It uses the stack in many places to avoid unnecessary allocations while processing events. They are not likely to need more memory. If we hit a NULL allocation, it's more likely because memory is leaking somewhere or we have memory fragmentation. Providing the file and line gives them something to report that someone can look into.

**Comment by Yugend** [comment]
> Providing the file and line gives them something to report that someone can look into.

Then it really makes sense, I didn't think about it.
Added two commits to my pull request. I took into account your recommendations about error messages and tried to fix #362 opened by myself.

File: src/auditctl.c [hunk 2]
```diff
@@ -1391,6 +1391,11 @@ static int fileopt(const char *file)
 		}
 		i = 0;
 		fields = malloc(nf * sizeof(char *));
+		if (fields == NULL) {
+			audit_msg(LOG_ERR, "Out of memory. Check %s file, %d line", __FILE__, __LINE__);
+			return 1;
```
Comment (Cropi) [inline src/auditctl.c:1396]: I would suggest to close the file descriptor before leaving the function, via `fclose(f);`

**Review by Cropi — CHANGES_REQUESTED** [review]

**Comment by Yugend** [comment]
Thank you for your advices, guys, I really appreciate it. I am not a very experienced coder, including in git. I'm sorry to give you so much trouble with my PR.

**Comment by stevegrubb** [comment]
Hello, I was looking to apply this patch, but it has 2 problems. It patches ipctab.h and auditd.c which were changed on the master branch while this patch was being made. I don't know how it got added to the patch, but we need to drop that portion of the patch. Revert the changes to those 2 files. Or if you allow editing of the PR by the maintainers, I might be able to fix it.

**Comment by stevegrubb** [comment]
It was applied as a series of commits. The last of which is commit 3f3b3a2. Thanks for the patch!

**Comment by Yugend** [comment]
Maybe it's my fault, I clicked somewhere wrong when I saw a message about differences in the master branch and my PR or something like that.
Anyway, thank you for the feedback!

**Pull request closed without merging**